etl/helpers.py

```diff
@@ -729,6 +726,21 @@ def check_that_all_active_steps_are_necessary(self) -> None:
         if len(unused_data_steps) > 0:
             log.warning(f"Some data steps can be safely archived: {unused_data_steps}")
 
+    def get_backported_db_dataset_ids(self) -> List[int]:
+        """Get list of ids of DB datasets that are used as backported datasets in active steps of ETL.
+
+        Returns
+        -------
+        backported_dataset_ids : List[int]
+            Grapher DB dataset ids that are used in ETL backported datasets.
+        """
+        backported_dataset_names = [step for step in self.all_active_dependencies if step.startswith("backport://")]
+        backported_dataset_ids = sorted(
+            set([int(step.split("dataset_")[1].split("_")[0]) for step in backported_dataset_names])
```

Comment:
I'd lean towards having a dedicated function can recognize the different fields from a backported dataset URI. For instance

```python
def extract_dataset_id_from_backported_dataset(dataset_uri: str) -> int:
    rex = r"backport://backport/(?P<namespace>[a-z]+)/(?P<version>[\d+\-]|latest)/dataset_(?P<dataset_id>\d+)_(?P<dataset_slug>[a-z_\d]+)"
    match = re.match(rex, dataset_uri)
    if match:
        parameters = match.groupdict()
        # return id
        try:
            dataset_id = int(parameters["dataset_id"])
        except ValueError:
            raise ValueError(f"Invalid dataset id `{parameters['dataset_id']}`!")
        return dataset_id
    raise SyntaxError(f"Unrecognized backported dataset URI `{dataset_uri}`.")
```

And then call it from here:

```python
backported_dataset_ids = sorted(
            set(extract_dataset_id_from_backported_dataset(step) for step in backported_dataset_names)
```

Also, note that you do not need the brackets inside `set()`.

Reply:
Thanks for the suggestion! For this script it's not worth the effort of changing it (and I already have a subbranch, so I'd prefer to skip unnecessary changes). We can improve these functions if we move them to `VersionTracker`.